app/views/projects/_form.html.erb

```diff
@@ -20,9 +17,43 @@
   <%= text_field_with_label(form: f, field: :place_name,
                             placeholder: :start_typing.l,
                             label: :WHERE.t + ":",
-                            between: :required,
                             data: { autocompleter: :location }) %>
 
+  <%= tag.div(
+    tag.p("*#{:form_projects_date_range.t}*".t) +
+    tag.p(:form_projects_dates_explain.t)
+  ) %>
+  <div class="container-text ml-4">
+    <div>
+      <%= f.label(:start_date, "#{:form_projects_start_date.t}: ") %>
+      <%= f.date_select(:start_date,
+                        { class: "form-control mb-2",
+                          start_year: Date.today.year - 10,
+                          end_year: Date.today.year + 10 }) %>
+    </div>
+    <div>
+      <%= f.label(:end_date, "#{:form_projects_end_date.t}: ") %>
+      <%= f.date_select(:end_date,
+                        { class: "form-control mb-2",
+                          start_year: Date.today.year - 10,
+                          end_year: Date.today.year + 10 }) %>
+    </div>
+    <div>
+      <p>
+        <%= radio_button_with_label(
+              form: f, field: "dates_any", value: false, checked: !@project_dates_any,
+              label: " #{:form_projects_range.t} (#{:form_projects_use_date_range.t})"
+            ) %>
+      </p>
+      <p>
+        <%= radio_button_with_label(
```

Comment:
This is still not achieving what I was looking for where you can click on the text as well as the button itself.  I created something that I like better on `njw-wrestling-with-radio-buttons`.  In the process I realized that were both radio_with_label and radio_button_with_label in app/helpers/forms_helper.rb and the first actually does more of the right stuff in my view which I largely recreated in radio_button_with_label.  It does seem like we should only have one of these.  However, I still don't like how separated the radio buttons are in radio_with_label and I just ran out of time to push on it further.  I also haven't seen where else that method is called.

Reply:
@mo-nathan `radio_with_label` and `radio_button_with_label` are intentionally different helpers for UI variations that Bootstrap offers out of the can. 

There is no need to wrestle... all Bootstrap radios with labels (i.e. correctly formatted following their markup examples) have clickable text. The helper you want is `radio_with_label`. (In the markup, the nesting of elements is essential: the wrapping CSS class must be "checkbox", and the input must be inside the label. That's what the `radio_with_label` helper does for you so you don't have to remember.)

The `radio_button_with_label` wraps the **label** in Bootstrap "button" markup (i.e. the CSS class "btn btn-default" — think "submit button") and hides the "radio" circle-checkbox "button".  This is used for the "either-or" array of sorting buttons on index pages.
![Screen Shot 2023-12-02 at 2 28 10 PM](https://github.com/MushroomObserver/mushroom-observer/assets/1948095/9ea0b58e-35fe-4fab-af3f-bcc3812df5eb)

The indispensable, timesaving reference, woefully underconsulted on MO, is the Bootstrap docs themselves. These are the three different variations on a radio UI that they offered in version 3:
https://getbootstrap.com/docs/3.4/css/#checkboxes-and-radios
https://getbootstrap.com/docs/3.4/javascript/#buttons-checkbox-radio
https://getbootstrap.com/docs/3.4/components/#input-groups-checkboxes-radios

I think I wrote `radio_with_label` for completeness when i was writing the `checkbox_with_label`, so i'd have it on hand — when you're trying to DRY up hundreds of views you do things like that 😓, but i'm not surprised it isn't called anywhere.